src/pages/LoggedIn.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of the GNU Affero General Public License v3.0.
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useEffect } from 'react';
+import { css } from "emotion";
+import DNALoader from "@icgc-argo/uikit/DnaLoader";
+import { RouteComponentProps } from 'react-router-dom';
+
+import { ModalPortal } from "App";
+import { EGO_TOKEN_ENDPOINT } from 'config/globals';
+import { useAuth } from "providers/Auth";
+import { isValidJwt } from 'utils/egoJwt';
+import { getRedirectUrl, clearRedirectUrl } from "utils/redirectUrl";
+
+export default ({ history }: RouteComponentProps) => {
+  const { isLoggedIn, setToken } = useAuth();
+  
+  const redirect = () => {
+    const redirectUrl = getRedirectUrl();
+    
+    // TODO: add permission checking before redirect
+    if (redirectUrl) {
+      clearRedirectUrl();
+      history.push(redirectUrl);
+    } else {
+      history.push('/');
```

Comment:
something tells me `.replace` is better because. this page is really just an ephemeral step to somewhere else...

Maybe consider it for other places too :)

Reply:
Good catch! I'll update this and some of the other places where it makes more sense. Thanks :)